autogpt_platform/autogpt_libs/autogpt_libs/supabase_integration_credentials_store/store.py

```diff
@@ -108,8 +108,13 @@ async def store_state_token(
     async def get_scopes_from_state_token(
         self, user_id: str, token: str, provider: str
     ) -> list[str]:
+        """
+        Get the valid scopes from the OAuth state token.
+        """
         user_metadata = self._get_user_metadata(user_id)
-        oauth_states = user_metadata.get("integration_oauth_states", [])
+        oauth_states = user_metadata.get("integration_oauth_states")
+        if not oauth_states:
+            raise ValueError(f"No OAuth states found for user with ID {user_id}")
```

Comment:
Maybe this comment is premature but I think what you meant to do is (also) add this check for `valid_state`, so it raises if the requested state doesn't exist or is expired

Reply:
Our premise for this was wrong, we do want that to return empty and this to not error. The assumption here that we return [] means we need to renew the tokens. I've renamed the function to be more clear on its usage